pkg/controllers/build/container_controller.go

```diff
@@ -129,8 +158,74 @@ func (r *ContainerReconciler) resolveTargetImage(ctx context.Context, log logr.L
 	return image, nil
 }
 
+func (r *ContainerReconciler) resolveDigestReference(ctx context.Context, log logr.Logger, ref name.Reference, container *buildv1alpha1.Container) (string, error) {
+	keychain, err := r.constructKeychain(ctx, log, container)
+	if err != nil {
+		return "", err
+	}
+
+	auth, err := keychain.Resolve(ref.Context().Registry)
+	if err != nil {
+		log.Error(err, "unable to resolve auth for registry", "registry", ref.Context().RegistryStr())
+		return "", err
+	}
+
+	img, err := remote.Image(ref, remote.WithAuth(auth))
+	if err != nil {
+		log.Error(err, "failed to read image", "image", ref.String())
+		return "", err
+	}
+
+	digest, err := img.Digest()
+	if err != nil {
+		log.Error(err, "failed to get image digest", "image", ref.String())
+		return "", err
+	}
+
+	return fmt.Sprintf("%s@%s", ref.Context().Name(), digest), nil
```

Comment:
Will this preserve any tag from the input reference? Since `ref.Context()` returns a _repository_ reference, I guess that would strip out any tag.

Reply:
@glyn It does strip out the tag. `ref.Name()` includes the tag for a tag reference but `ref.Context().Name()` shouldn't include the tag or digest.

Reply:
Ok, so the question is: is that what is required? It's architecturally ok to have a tag plus a digest (although some tooling can't cope with it) and sometimes preserving the digest is helpful for human readability.

Reply:
What gets set on `LatestImage` with the given implementation should be a digest reference w/o a tag which seems to me to be safer choice (given that as you mention some tooling doesn't like having both a tag and a digest).